executions/graphql-kotlin-automatic-persisted-queries/src/main/kotlin/com/expediagroup/graphql/apq/cache/AutomaticPersistedQueriesCache.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.apq.cache
+
+import graphql.ExecutionInput
+import graphql.execution.preparsed.PreparsedDocumentEntry
+import graphql.execution.preparsed.persisted.PersistedQueryCache
+import graphql.execution.preparsed.persisted.PersistedQueryCacheMiss
+import java.util.concurrent.CompletableFuture
+
+interface AutomaticPersistedQueriesCache : PersistedQueryCache {
+
+    @Deprecated(
+        message = "deprecated in favor of async retrieval of PreparsedDocumentEntry",
+        replaceWith = ReplaceWith("getPersistedQueryDocumentAsync(persistedQueryId, executionInput, onCacheMiss)")
+    )
+    override fun getPersistedQueryDocument(
```

Comment:
why are we adding a deprecated method?

Reply:
The `PersistedQueryCache` from `graphql-java` marks `getPersistedQueryDocument` as deprecated, so we are forced to implement it. 
All in favor of `getPersistedQueryDocumentAsync`